Lesson07.Strings/Lesson07.Strings/Program.cs

```diff
@@ -63,27 +64,114 @@ static void Main(string[] args)
 
             var emptyString = string.Empty;
             const int N = 1000;
-            for(int i = 0; i < N; i++)
+            for (int i = 0; i < N; i++)
             {
                 emptyString += $"{i} ";
             }
 
             var stringBuilder = new StringBuilder();
 
-            for(int j = 0;j < N; j++)
+            for (int j = 0; j < N; j++)
             {
                 stringBuilder.AppendFormat("{0} ", j);
             }
             Console.WriteLine(emptyString);
             Console.WriteLine(stringBuilder.ToString());
 
             //abcbcd => ["b", "c"]
+            //Compare
+            Console.WriteLine("Please enter first string: ");
+            string first = Console.ReadLine();
+            Console.WriteLine($"Your string is: {first}");
+            Console.WriteLine("Please enter second string: ");
+            string second = Console.ReadLine();
+            Console.WriteLine($"Your string is {second}");
+            Console.WriteLine($"Method Compare is: {Compare(first, second)}");
+            //Analyze 
+            Console.WriteLine($"{Environment.NewLine}Please enter string: ");
+            first = Console.ReadLine();
+            Console.WriteLine($"Your string is: {first} {Environment.NewLine} Method Analyze is: {Analyze(first)}");
+            //Sort
+
+            Console.WriteLine($"{Environment.NewLine}Please enter string for sorting: ");
+            first = Console.ReadLine();
+            Console.WriteLine($"Your string is: {first} {Environment.NewLine}Method Sort is: {Sort(first)}");
+
+            //Duplicate
+            Console.WriteLine($"{Environment.NewLine}Please enter string for duplicate: ");
+            first = Console.ReadLine();
+            string result = Duplicate(first);
+            Console.WriteLine($"Result method Duplicate is: {result}");
         }
 
         private void Print(string placeholder, string name)
         {
             Console.WriteLine(string.Format(placeholder, name));
             Console.WriteLine($"Hello, {name}");
         }
+
+        //homework
+        private static bool Compare(string first, string second)
+        {
+            bool result = false;
+            for (int i = 0; i < first.Length; i++)
+            {
+                for (int j = 0; j < second.Length; j++)
+                {
+                    if (first.Length == second.Length && first[i] == second[j])
+                        result = true;
```

Comment:
you can `return true` directly from here. It doesn't make any sense to continue a loop.